mm2src/coins/eth/v2_activation.rs

```diff
@@ -937,3 +937,209 @@ fn compress_public_key(uncompressed: H520) -> MmResult<H264, EthActivationV2Erro
     let compressed = public_key.serialize();
     Ok(H264::from(compressed))
 }
+
+/// Eth from coin conf v2 activation for `eth_docker_tests.rs`
+/// It uses special `build_web3_instances_for_test` function
+#[cfg(all(feature = "for-tests", not(target_arch = "wasm32")))]
```

Comment:
Why not use this conditional compilation attribute in the original `eth_coin_from_conf_and_request_v2` function and add `build_web3_instances_for_test` under it? We don't have to maintain 2 functions this way in case we changed the original one.

Reply:
> Why not use this conditional compilation attribute in the original eth_coin_from_conf_and_request_v2 function and add build_web3_instances_for_test under it?

![Screenshot 2024-08-25 at 10 08 20](https://github.com/user-attachments/assets/e42c9e05-6ab4-4b10-af86-fb867b8bee9a)

it will lead to clippy errors
```
error: field `komodo_proxy` is never read
  --> mm2src/coins/eth/web3_transport/http_transport.rs:49:16
   |
47 | pub struct HttpTransportNode {
   |            ----------------- field in this struct
48 |     pub(crate) uri: http::Uri,
49 |     pub(crate) komodo_proxy: bool,
   |                ^^^^^^^^^^^^
   |
   = note: `HttpTransportNode` has derived impls for the traits `Debug` and `Clone`, but these are intentionally ignored during dead code analysis
   = note: `-D dead-code` implied by `-D warnings`

error: function `build_web3_instances` is never used
   --> mm2src/coins/eth/v2_activation.rs:791:10
    |
791 | async fn build_web3_instances(
    |          ^^^^^^^^^^^^^^^^^^^^

error: could not compile `coins` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error: could not compile `coins` (lib test) due to 2 previous errors
```

as in production we dont use `"for-tests"` feature

Reply:
We can add in `eth_coin_from_conf_and_request_v2` doc comment  "not to forget to change eth_coin_from_conf_and_request_v2_for_test function" if you did changes. same for `build_web3_instances_for_test`.

Are you ok with the suggestion above? we shouldn't add `#[allow(dead_code)]` annotation to `build_web3_instances` function, as clippy should check it.

Reply:
added doc comms with notes for now https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/commits/f1e6b20218111e53736627ad762ebc8cf29c40a8

Reply:
Simplified `build_web3_instances` here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/commits/f9f8d50ccab5180d1ee86e62d5c4b5745e469513 p2p ctx is only used if `komodo_proxy`  is true. Please check the code to make sure I didn't make any mistakes.

Reply:
> p2p ctx is only used if komodo_proxy is true

oh, exactly! right now `build_web3_instances` looks beautiful. everything is fine.